src/HydraAuctionOnchain/Helpers.hs

```diff
@@ -88,6 +89,18 @@ pfindUniqueInputWithScriptHash = phoistAcyclic $
       #$ pfield @"inputs"
       # txInfo
 
+pfindUniqueRefInputWithScriptHash :: Term s (PScriptHash :--> PTxInfo :--> PMaybe PTxInInfo)
+pfindUniqueRefInputWithScriptHash = phoistAcyclic $
+  plam $ \sh txInfo ->
+    pfindUnique
+      # plam
+        ( \utxo -> P.do
+            addr <- plet $ pfield @"address" #$ pfield @"resolved" # utxo
+            paddressHasScriptHash # addr # sh
+        )
+      #$ pfield @"referenceInputs"
+      # txInfo
+
```

Comment:
I think projects should try to upstream functions like that if possible. It's a shame we don't have all that stuff in plutarch-plutus.